lib/web_ui/lib/src/ui/window.dart

```diff
@@ -638,6 +643,36 @@ abstract class Window {
   Brightness get platformBrightness => _platformBrightness;
   Brightness _platformBrightness = Brightness.light;
 
+  /// Reference to css media query that indicates the user theme preference on the web
+  final html.MediaQueryList _brightnessMediaQuery = html.window.matchMedia('(prefers-color-scheme: dark)');
+
+  /// A callback that is invoked whenever [_brightnessMediaQuery] changes value.
+  /// Updates the platformBrightness with the new user preferenceq
+  ///
+  ///  The param brightnessMediaQuery can be a MediaQueryListEvent or a MediaQueryList;
+  ///
+  void _brightnessMediaQueryChanged(brightnessMediaQuery) {
+      Brightness previousPlatformBrightness = _platformBrightness;
+      _platformBrightness = brightnessMediaQuery.matches ? Brightness.dark : Brightness.light;
+      if (previousPlatformBrightness != _platformBrightness &&
+          _onPlatformBrightnessChanged is VoidCallback)
+        _onPlatformBrightnessChanged();
+  }
+
+  /// Set the callback function for listening changes in [_brightnessMediaQuery] value
+  ///
+  void _addBrightnessMediaQueryListener() {
+    _brightnessMediaQueryChanged(_brightnessMediaQuery); // Brightness is checked when initializing
+    _brightnessMediaQuery.addListener(_brightnessMediaQueryChanged);
+  }
+
+  /// Remove the callback function for listening changes in [_brightnessMediaQuery] value
+  ///
+  void _removeBrightnessMediaQueryListener() {
+    _brightnessMediaQuery.removeListener(_brightnessMediaQueryChanged);
```

Comment:
I think this won't properly remove the listener because `_brightnessMediaQueryChanged` is a tear off that creates a new function every time. Is that right, @yjbanov?

Reply:
I think tear-offs work correctly in this context, since this should implement `==`. Double check in a test to be sure?

Reply:
That's right. Dart autobinds `this` which causes a new function object to be allocated on every tear-off. You'll want to do something like this: https://github.com/flutter/engine/blob/master/lib/web_ui/lib/src/engine/semantics/incrementable.dart#L64